optimum/habana/diffusers/models/unet_spatio_temporal_condition.py

```diff
@@ -0,0 +1,155 @@
+from typing import Tuple, Union
+
+import torch
+from diffusers.models.unets.unet_spatio_temporal_condition import UNetSpatioTemporalConditionOutput
+
+from optimum.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+def gaudi_unet_spatio_temporal_condition_model_forward(
+    self,
+    sample: torch.FloatTensor,
+    timestep: Union[torch.Tensor, float, int],
+    encoder_hidden_states: torch.Tensor,
+    added_time_ids: torch.Tensor,
+    return_dict: bool = True,
+) -> Union[UNetSpatioTemporalConditionOutput, Tuple]:
+    r"""
+    The [`UNetSpatioTemporalConditionModel`] forward method.
+
+    Args:
+        sample (`torch.FloatTensor`):
+            The noisy input tensor with the following shape `(batch, num_frames, channel, height, width)`.
+        timestep (`torch.FloatTensor` or `float` or `int`): The number of timesteps to denoise an input.
+        encoder_hidden_states (`torch.FloatTensor`):
+            The encoder hidden states with shape `(batch, sequence_length, cross_attention_dim)`.
+        added_time_ids: (`torch.FloatTensor`):
+            The additional time ids with shape `(batch, num_additional_ids)`. These are encoded with sinusoidal
+            embeddings and added to the time embeddings.
+        return_dict (`bool`, *optional*, defaults to `True`):
+            Whether or not to return a [`~models.unet_slatio_temporal.UNetSpatioTemporalConditionOutput`] instead of a plain
+            tuple.
+    Returns:
+        [`~models.unet_slatio_temporal.UNetSpatioTemporalConditionOutput`] or `tuple`:
+            If `return_dict` is True, an [`~models.unet_slatio_temporal.UNetSpatioTemporalConditionOutput`] is returned, otherwise
+            a `tuple` is returned where the first element is the sample tensor.
+    """
```

Comment:
Can you please put comment that where the original code came from and what hpu specific change added here?

Reply:
I will add comment on source and HPU changes

Reply:
I removed modified forward unet call from PR as its obsolete (WA for 1.11 not needed anymore)